compilers/concrete-compiler/compiler/lib/Common/Transformers.cpp

```diff
@@ -184,9 +184,51 @@ Result<TransportValueVerifier> getObliviousTransportValueVerifier() {
   };
 }
 
-Result<TransportValueVerifier>
-getTransportValueVerifier(const Message<concreteprotocol::GateInfo> &gateInfo) {
+Message<concreteprotocol::GateInfo>
+updateGateInfoAccordingValue(Message<concreteprotocol::GateInfo> &gate,
+                             const TransportValue &value) {
+
+  auto gateReader = gate.asReader();
+  auto gateTypeInfo = gateReader.getTypeInfo();
+  if (!gateTypeInfo.hasLweCiphertext()) {
+    return gate;
+  }
+  auto gateCiphertext = gateTypeInfo.getLweCiphertext();
+  auto gateCompression = gateCiphertext.getCompression();
+  if (gateCompression == concreteprotocol::Compression::SEED) {
+    auto valueReader = value.asReader();
+    auto valueTypeInfo = valueReader.getTypeInfo();
+    if (!valueTypeInfo.hasLweCiphertext()) {
+      return gate;
+    }
+    auto valueCiphertext = valueTypeInfo.getLweCiphertext();
+    auto valueCompression = valueCiphertext.getCompression();
+    if (valueCompression == concreteprotocol::Compression::NONE) {
+      // If the compression of transportValue is none and the gateInfo have
+      // compression we update the gateInfo to allow uncompressed transportValue
+      auto gateBuilder = gate.asBuilder();
+      gateBuilder.getTypeInfo().getLweCiphertext().setCompression(
+          concreteprotocol::Compression::NONE);
+      auto gateDimensions = gateBuilder.getRawInfo().getShape().getDimensions();
+      auto lweSize = gateCiphertext.getEncryption().getLweDimension() + 1;
+      gateDimensions.set(gateDimensions.size() - 1, lweSize);
+      auto concreteShapeDimensions = gateBuilder.getTypeInfo()
+                                         .getLweCiphertext()
+                                         .getConcreteShape()
+                                         .getDimensions();
+      concreteShapeDimensions.set(concreteShapeDimensions.size() - 1, lweSize);
```

Comment:
I don't think it is valid to just append in the list like this. I think you have to re init the field with the proper size.

Reply:
Its not an append, I change the last dimension which was the size of compressed ciphertext by the size of uncompressed ciphertext.